src/Functions/max2.cpp

```diff
@@ -0,0 +1,22 @@
+#include <Functions/FunctionMathBinaryFloat64.h>
+#include <Functions/FunctionFactory.h>
+
+template <typename T>
+T max(T a, T b)
+{
+    return a > b ? a : b;
+}
+
+namespace DB
+{
+namespace
+{
+    struct Max2Name { static constexpr auto name = "max2"; };
+    using FunctionMax2 = FunctionMathBinaryFloat64<BinaryFunctionVectorized<Max2Name, max>>;
```

Comment:
> Maybe should be updated as there is a conversion to Float64, as I can understand.

Yes it returns Fload64 for all input types.

```
SELECT toTypeName(max2(2, 1))

┌─toTypeName(max2(2, 1))─┐
│ Float64                │
└────────────────────────┘
```

Reply:
Thanks. I think that it is not good to deduce such an exact function to the `Float64` type, but do not know a convenient way to fix it.